docs/topics/exceptions.md

```diff
@@ -1,122 +1,679 @@
+        count()
+        
+    } catch (e: ArithmeticException) {
+        
+        // If count() throws an exception, the catch block returns -1, which is assigned to num
+        -1
+    }
+    println("Result: $num")
+}
 
-`try` is an expression, which means it can have a return value:
+// Simulates a function that might throw ArithmeticException
+fun count(): Int {
+    
+    // Change this value to return a different value to num
+    val a = 0
+    
+    return 10 / a
+}
+```
+{kotlin-runnable=”true”}
+
+
+Kotlin supports using several handlers associated with the same `try` block, each designed to catch only one specific type of exception. 
+You can add as many `catch` blocks as needed to handle different exceptions distinctively.
+
+Let’s look at an example using [custom exceptions](#creating-custom-exceptions):
 
 ```kotlin
-val a: Int? = try { input.toInt() } catch (e: NumberFormatException) { null }
+open class WithdrawalException(message: String) : Exception(message)
+class InsufficientFundsException(message: String) : WithdrawalException(message)
+
+fun processWithdrawal(amount: Double, availableFunds: Double) {
+    if (amount > availableFunds) {
+        throw InsufficientFundsException("Insufficient funds for the withdrawal.")
+    }
+    if (amount < 1 || amount % 1 != 0.0) {
+        throw WithdrawalException("Invalid withdrawal amount.")
+    }
+    println("Withdrawal processed")
+}
+
+fun main() {
+    val availableFunds = 500.0
+
+    // Change this value to test different scenarios
+    val withdrawalAmount = 500.5
+
+    try {
+        processWithdrawal(withdrawalAmount.toDouble(), availableFunds)
+
+    // The order of catch blocks is important!
+    } catch (e: InsufficientFundsException) {
+        println("Caught an InsufficientFundsException: ${e.message}")
+    } catch (e: WithdrawalException) {
+        println("Caught a WithdrawalException: ${e.message}")
+    }
+}
 ```
+{kotlin-runnable="true"}
 
-The returned value of a `try` expression is either the last expression in the `try` block or the
-last expression in the `catch` block (or blocks).
-The contents of the `finally` block don't affect the result of the expression.
+When you have multiple `catch` blocks, it's important to order them from the most specific to the least specific exception,
+following a top-to-bottom order in your code. 
+This ordering aligns with the program's execution flow. 
+If you were to catch a general exception before a more specific one, the specific exception would never be reached, 
+as the general catch block would intercept all exceptions, including the specific ones.
 
-## Checked exceptions
+### The finally block
 
-Kotlin does not have checked exceptions. There are many reasons for this, but we will provide a simple example that illustrates why it is the case.
+The `finally` block is a block of code that is always executed, regardless of whether the `try` block succeeds without
```

Comment:
Hmm 🤔 - I might prefer to start with the object here as I'd like to put it in the focus instead of the action - how about this version?
The `finally` block contains code that always executes, regardless of whether the `try` block completes successfully or throws an exception.